src/components/transactions/TxList/index.tsx

```diff
@@ -1,29 +1,62 @@
 import GroupedTxListItems from '@/components/transactions/GroupedTxListItems'
-import { groupConflictingTxs } from '@/utils/tx-list'
+import BulkTxListGroup from '@/components/transactions//BulkTxListGroup'
+import { groupTxs } from '@/utils/tx-list'
 import { Box } from '@mui/material'
-import type { TransactionListPage } from '@safe-global/safe-gateway-typescript-sdk'
+import type { Transaction, TransactionDetails, TransactionListPage } from '@safe-global/safe-gateway-typescript-sdk'
 import type { ReactElement, ReactNode } from 'react'
 import { useMemo } from 'react'
 import TxListItem from '../TxListItem'
 import css from './styles.module.css'
+import useAsync from '@/hooks/useAsync'
+import useChainId from '@/hooks/useChainId'
+import { isTransactionListItem } from '@/utils/transaction-guards'
+import { getTxDetailsWithBackoff } from '@/utils/transactions'
+import { uniq } from 'lodash'
 
 type TxListProps = {
   items: TransactionListPage['results']
 }
 
+const getBulkGroupTxHash = (group: Transaction[], txHashes: Record<string, string>) => {
+  const hashList = group.map((item) => {
+    const txId = item.transaction.id
+    return txHashes[txId]
+  })
+  return uniq(hashList).length === 1 ? hashList[0] : undefined
+}
+
 export const TxListGrid = ({ children }: { children: ReactNode }): ReactElement => {
   return <Box className={css.container}>{children}</Box>
 }
 
 const TxList = ({ items }: TxListProps): ReactElement => {
-  const groupedItems = useMemo(() => groupConflictingTxs(items), [items])
+  const chainId = useChainId()
+
+  const [txHashesById] = useAsync(async () => {
+    const transactions = items.filter(isTransactionListItem)
+    return await getTxDetailsWithBackoff(transactions, chainId).then((txDetailsList) => {
+      return txDetailsList.reduce((accumulator: Record<string, string>, txDetails: TransactionDetails) => {
+        const { txId, txHash } = txDetails
+        if (!txId || !txHash) return accumulator
+        accumulator[txId] = txHash
+        return accumulator
+      }, {})
+    })
+  }, [chainId, items])
```

Comment:
This is an impactful change as we would now fetch txDetails for every transaction in the history whenever the user opens the page and/or scrolls. If I understand correctly all thats missing is the `txHash` of each transaction?

Reply:
yeah it would be much better to get the tx hash /history if that's possible

Reply:
> As far as I can see the `/all-transactions` endpoint on the transaction service returns the `txHash`. @iamacook is it possible to also return it from the `/history` endpoint on the CGW?

It is and whilst it makes sense to return it in the summary, I'm not sure why it isn't. I'll bring it up for discussion next week.

On a separate note: you might want to consider ensuring the transaction has execution details before grouping under "bulk execution". I'm not certain but I think the current implementation may group transfers that occur in the same transaction.

Reply:
It's updated now to use the changes from the CGW. Cheer Aaron 👍